sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/batch/BatchResponseParser.java

```diff
@@ -0,0 +1,245 @@
+                // batch request is successful - so fail as InternalServerError if this is not the case.
+                response = new TransactionalBatchResponse(
+                    HttpResponseStatus.INTERNAL_SERVER_ERROR.code(),
+                    0,
+                    "Invalid server response",
+                    documentServiceResponse.getResponseHeaders(),
+                    documentServiceResponse.getCosmosDiagnostics(),
+                    request.getOperations());
+            }
+
+            // When the overall response status code is TooManyRequests, propagate the RetryAfter into the individual operations.
+            int retryAfterMilliseconds = 0;
+
+            if (responseStatusCode == HttpResponseStatus.TOO_MANY_REQUESTS.code()) {
+                String retryResponseValue = documentServiceResponse.getResponseHeaders().getOrDefault(RETRY_AFTER_IN_MILLISECONDS, null);
+                if (StringUtils.isNotEmpty(retryResponseValue)) {
+                    try {
+                        retryAfterMilliseconds = Integer.parseInt(retryResponseValue);
+                    } catch (NumberFormatException ex) {
+                        // Do nothing. It's number format exception
+                    }
+                }
+            }
+
+            response.createAndPopulateResults(request.getOperations(), retryAfterMilliseconds);
+        }
+
+        checkState(response.size() == request.getOperations().size(),
+            "Number of responses should be equal to number of operations in request.");
+
+        return Mono.just(response);
+    }
+
+    private static TransactionalBatchResponse populateFromResponseContentAsync(
+        final RxDocumentServiceResponse documentServiceResponse,
+        final ServerBatchRequest request,
+        final boolean shouldPromoteOperationStatus) {
+
+        final ArrayList<TransactionalBatchOperationResult<?>> results = new ArrayList<>();
+        final String responseContent = documentServiceResponse.getResponseBodyAsString();
+
+        if (responseContent.charAt(0) != HYBRID_V1) {
+            // Read from a json response body. To enable hybrid row just complete the else part
+            final ObjectMapper mapper = Utils.getSimpleObjectMapper();
+
+            try{
+                final ObjectNode[] objectNodes = mapper.readValue(documentServiceResponse.getResponseBodyAsString(), ObjectNode[].class);
+                for (ObjectNode objectInArray : objectNodes) {
+                    final TransactionalBatchOperationResult<?> batchOperationResult = BatchResponseParser.createBatchOperationResultFromJson(objectInArray);
+                    results.add(batchOperationResult);
+                }
+            } catch (IOException ex) {
+                logger.error("Exception in parsing response", ex);
+            }
+
+        } else {
+            // TODO(rakkuma): Implement hybrid row response parsing logic here. Parse the response hybrid row buffer
```

Comment:
Please file a github work item to track this work and add the item number here / link here.

Reply:
Created and added the workitem. Thanks.